### ISSUE TITLE
Facet clearing functionality relies on code outside of module

### ISSUE THREAD

**Issue by fantapop** [eucalyptus/magic-search#16]
the function `removeFacet` clears `$scope.currentSearch` and then calls `$scope.initFacets` which resyncs currentSearch from the url.

magic_search.js does not add the search to the url so when you clear try to clear an individual facet all facets are removed.  The demo adds an additional javascript file main.js which is not included with the module that updates the url after each facet is added.  The url update could should be handled inside the module.

**Comment by dkavanagh** [comment]
I agree. I've also dealt with this and just put up with it till now. I'm opening another issue for a greater re-factor I've had in mind for a while now (since around ng-conf time).
